waiter/src/waiter/reporter.clj

```diff
@@ -0,0 +1,151 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns waiter.reporter
+  (:require [clj-time.core :as t]
+            [metrics.core :as metrics]
+            [schema.core :as s]
+            [waiter.schema :as schema]
+            [clojure.tools.logging :as log])
+  (:import (com.codahale.metrics ConsoleReporter MetricFilter ScheduledReporter)
+           (com.codahale.metrics.graphite Graphite GraphiteReporter GraphiteSender PickledGraphite)
+           java.io.PrintStream
+           java.net.InetSocketAddress
+           java.util.concurrent.TimeUnit))
+
+(defprotocol CodahaleReporter
+  "A reporter for codahale metrics"
+
+  (close! [this]
+    "Stops the reporter and performs any cleanup")
+
+  (state [this]
+    "Returns the state of this reporter"))
+
+(defn- make-metrics-filter
+  [filter-regex]
+  (reify MetricFilter (matches [_ name _] (some? (re-matches filter-regex name)))))
+
+(defn- make-codahale-reporter
+  [^ScheduledReporter scheduled-reporter state period-ms]
+  (.start scheduled-reporter period-ms TimeUnit/MILLISECONDS)
+  (reify CodahaleReporter
+    (close! [_] (.close scheduled-reporter) (reset! state :closed))
```

Comment:
If you like the atom initialization suggested below. we can change it to `(swap! state assoc :state :closed)`.